stratosphere/ncm/source/lr_redirectonlylocationresolver.cpp

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2019 Adubbz
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "impl/ncm_content_manager.hpp"
+#include "lr_redirectonlylocationresolver.hpp"
+
+namespace sts::lr {
+
+    RedirectOnlyLocationResolverInterface::~RedirectOnlyLocationResolverInterface() {
+        this->program_redirector.ClearRedirections();
+        this->debug_program_redirector.ClearRedirections();
+        this->app_control_redirector.ClearRedirections();
+        this->html_docs_redirector.ClearRedirections();
+        this->legal_info_redirector.ClearRedirections();
+    }
+
+    Result RedirectOnlyLocationResolverInterface::ResolveProgramPath(OutPointerWithServerSize<Path, 0x1> out, ncm::TitleId tid) {
+        Path path;
+        
+        if (this->program_redirector.FindRedirection(&path, tid)) {
```

Comment:
Generally, I'm gonna stop commenting on this file, but use all the same feedback as for ContentLocationResolver.